lib/matplotlib/cm.py

```diff
@@ -663,3 +678,11 @@ def changed(self):
     *vmin*/*vmax* when a *norm* instance is given (but using a `str` *norm*
     name together with *vmin*/*vmax* is acceptable).""",
 )
+
+
+def _normalize_cmap(cmap):
```

Comment:
I was a bit confused with the name "normalize" here at first. I'm not sure I have a better one, but when you're also dealing with "resampling" in the PR it threw me off. Maybe `_get_cmap_or_default`?

Reply:
I tried `_ensure_cmap` because it will also short-circuit on being passed a `Colormap` object.